### ISSUE TITLE
ESMM 模型中 forward 部分在处理输出时是否有误？

### ISSUE THREAD

**Issue by neonx0826-cell** [yuangh-x/2022-NIPS-Tenrec#37]
由于我的设备无法跑全量 `ctr_data_1M` 数据集，因此我基于 `user_id` 对数据集进行了采样，仅保留了 **10%** 的数据用于实验。

## 原始实现

原代码中 `forward` 的实现如下：

```python
if self.num_task == 2:
    task_outputs[1] = task_outputs[0] * task_outputs[1]
    return task_outputs
```

但是模型训练时会出现‘顾此失彼’，测试集结果 click 的 AUC 很低（接近0.5），like 的 AUC 与论文中的结果相近。对 loss2 设置加权系数后，AUC 结果又反过来了。

## 后续修改
我根据 ESMM 论文修改 forward 输出部分如下：
```python       
if self.num_task == 2:
    p_ctr = torch.sigmoid(task_outputs[0])
    p_cvr = torch.sigmoid(task_outputs[1])
    task_outputs[1] = p_ctr * p_cvr
    return task_outputs
```
并修改了 mtltrain 中的 loss 部分如下：
```python 
...

loss_function_ctr = nn.BCEWithLogitsLoss()
loss_function_ctcvr = nn.BCELoss()
...

loss_1 = loss_function_ctr(predict[0], y1.unsqueeze(1).float())
loss_2 = loss_function_ctcvr(predict[1], y1.unsqueeze(1).float()*y2.unsqueeze(1).float())
loss = loss_1 + loss_2
```
之后的测试集结果与论文相近。请问这种现象的产生是由于采样导致的数据分布偏差，还是原代码输出处理有误？
